samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,413 @@
+			cellular_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_INTERVAL:
+			interval = atoi(optarg);
+			gnss_interval_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_GNSS_ACCURACY:
+			if (strcmp(optarg, "low") == 0) {
+				gnss_accuracy = LOC_ACCURACY_LOW;
+			} else if (strcmp(optarg, "normal") == 0) {
+				gnss_accuracy = LOC_ACCURACY_NORMAL;
+			} else if (strcmp(optarg, "high") == 0) {
+				gnss_accuracy = LOC_ACCURACY_HIGH;
+			} else {
+				shell_error(shell, "Unknown GNSS accuracy. See usage:");
+				goto show_usage;
+			}
+			gnss_accuracy_set = true;
+			break;
+		case LOCATION_SHELL_OPT_GNSS_NUM_FIXES:
+			gnss_num_fixes = atoi(optarg);
+			gnss_num_fixes_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_1:
+			if (strcmp(optarg, "cellular") == 0) {
+				method1 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method1 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method1_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_2:
+			if (strcmp(optarg, "cellular") == 0) {
+				method2 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method2 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method2_set = true;
+			break;
+		case '?':
+			goto show_usage;
+			break;
+		default:
+			shell_error(shell, "Unknown option. See usage:");
+			goto show_usage;
+		}
+	}
+
+	/* Handle location subcommands */
+	switch (loc_cmd_args.command) {
+	case LOCATION_CMD_CANCEL:
+		ret = location_request_cancel();
+		if (ret) {
+			shell_error(shell, "Cannot cancel location request, err: %d", ret);
+			return -1;
+		}
+		shell_print(shell, "Getting of location cancelled");
+		break;
+
+	case LOCATION_CMD_GET: {
+		struct loc_config config = { 0 };
+		struct loc_method_config methods[LOC_MAX_METHODS] = { 0 };
+		struct loc_config *real_config = &config;
+
+		config.interval = interval;
+		config.methods_count = method_count;
+		config.methods = methods;
+
+		if (method_count == 1) {
+			if (method1_set) {
+				methods[0].method = method1;
+			} else if (method2_set) {
+				methods[0].method = method2;
+			}
+		} else if (method_count == 2) {
+			if (method1_set) {
+				methods[0].method = method1;
+			}
+			if (method1_set && method2_set) {
+				methods[1].method = method2;
+			}
+			if (!method1_set && method2_set) {
+				methods[0].method = method2;
+			}
+		} else if (method_count != 0) {
+			shell_error(shell, "Max number of methods is %d.", LOC_MAX_METHODS);
+			goto show_usage;
+		} else {
+			/* No methods given. Use NULL to indicate default configuration. */
+			real_config = NULL;
+		}
+
+		if (methods[0].method == LOC_METHOD_GNSS) {
+			if (gnss_timeout_set) {
+				methods[0].gnss.timeout = gnss_timeout;
+			} else {
+				shell_error(shell, "gnss_timeout is mandatory to be given.");
```

Comment:
I guess `gnss_timeout `and `gnss_accuracy `could be optional

Reply:
Agreed

Reply:
Done